src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -199,7 +215,14 @@ class RegularSync(
     case ResponseReceived(peer, BlockBodies(blockBodies), timeTaken) =>
       log.info("Received {} block bodies in {} ms", blockBodies.size, timeTaken)
       waitingForActor = None
-      handleBlockBodies(peer, blockBodies)
+      matchBlockBodiesWithHeaders(peer, blockBodies).foreach { blocks =>
```

Comment:
it looks a bit strange, the logic is scattered. if the result is Some we handle the blocks here, if it's None we resume inside  `matchBlockBodiesWithHeaders`. How about moving side effects from this method here?

Reply:
This method handles various responses so I'd like to keep it's body short. I have another idea...